buildSrc/src/main/kotlin/otel.android-library-conventions.gradle.kts

```diff
@@ -18,13 +24,18 @@ android {
     }
 
     compileOptions {
-        val javaVersion = rootProject.extra["java_version"] as JavaVersion
         sourceCompatibility(javaVersion)
         targetCompatibility(javaVersion)
         isCoreLibraryDesugaringEnabled = true
     }
 }
 
+kotlin {
+    compilerOptions {
+        jvmTarget.set(JvmTarget.fromTarget(javaVersion.toString()))
+        languageVersion.set(minKotlinVersion)
```

Comment:
```suggestion
        languageVersion.set(minKotlinVersion)
        apiVersion.set(minKotlinVersion)
```

Reply:
Probably has to change in the other places as well

Reply:
Cool, I'll make the changes 👍 tbh the docs for that option weren't too clear for me so I wasn't sure if we needed it. Though I guess is better to have it to avoid unexpected issues. We should probably add it in the core repo too, as it's only got [languageVersion set](https://github.com/open-telemetry/opentelemetry-java/pull/5910).